Sources/Sentry/SentrySwizzle.m

```diff
@@ -106,10 +146,40 @@ @implementation SentrySwizzle
     pthread_mutex_lock(&gLock);
 
     originalIMP = class_replaceMethod(classToSwizzle, selector, newIMP, methodType);
+#if TEST || TESTCI
+    if (originalIMP) {
+        storeRefToOriginalImplementation(key, originalIMP);
+    }
+#endif // TEST || TESTCI
 
     pthread_mutex_unlock(&gLock);
 }
 
+#if TEST || TESTCI
+static void
+unswizzle(Class classToUnswizzle, SEL selector, const void *key)
+{
+    Method method = class_getInstanceMethod(classToUnswizzle, selector);
+
+    NSCAssert(NULL != method, @"Selector %@ not found in %@ methods of class %@.",
+        NSStringFromSelector(selector),
+        class_isMetaClass(classToUnswizzle) ? @"class" : @"instance", classToUnswizzle);
+
+    static pthread_mutex_t gLock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
`m`: If I'm not mistaken, this is a different mutex from the one in Swizzle. Wouldn't it be better to share one mutex as they manipulate the same dict.

Reply:
I agree, all the swizzle calls share the same mutex, moved it to the class level.